Yoast/wordpress-seo
# P2 301 only index on production

## Context

* When the result from wp_get_environment_type() is anything other than 'production', we shouldn't index our indexables.

## Summary

This PR can be summarized in the following changelog entry:

* Prevents indexing indexables outside of production.

## Relevant technical choices:

* Added Environment helper (also added $environment property to YoastSEO helper class).
* Added should_index_indexables method.
* Added a wpseo_override_save_indexables filter to all.
* Added a conditional "disabled" tag to the yoast-indexation warning on the SEO > Tools page

## Test instructions

This PR can be tested by following these steps:

- Use the Yoast_Test_Tools plugin to clear any current indexable data.

- Set your wordpress site to production mode; it should now be possible to navigate to SEO > Tools and press the button as shown in the screenshot:
![P2-301_button_enabled](https://user-images.githubusercontent.com/3604126/93106539-b041ca00-f6b0-11ea-9feb-46ad0a9b7446.png)

- Use the Yoast_Test_Tools plugin to clear indexable data.

- switch the wordpress site to a *staging* site, ( in other words, make ```\wpseo_functions\wp_get_environment_type``` return something other than 'production' )

- navigate to the Yoast SEO plugin menu. 

- Expected: The button should now be disabled and an additional warning should appear:
<img width="643" alt="P2-301_button_disabled" src="https://user-images.githubusercontent.com/3604126/93106520-a9b35280-f6b0-11ea-8da2-f404b49f672d.png">



## UI changes

## Documentation

## Quality assurance


Fixes https://yoast.atlassian.net/browse/P2-301
